waymo2argo/waymo_raw_data_to_argoverse.py

```diff
@@ -0,0 +1,441 @@
+                    log_id,
+                    ARGO_WRITE_DIR,
+                    track_id_dict,
+                )
+            if args.save_calibration:
+                calib_json = form_calibration_json(frame.context.camera_calibrations)
+                if log_calib_json is None:
+                    log_calib_json = calib_json
+                    calib_json_fpath = (
+                        f"{ARGO_WRITE_DIR}/{log_id}/vehicle_calibration_info.json"
+                    )
+                    check_mkdir(str(Path(calib_json_fpath).parent))
+                    save_json_dict(calib_json_fpath, calib_json)
+                else:
+                    assert calib_json == log_calib_json
+
+            # 5 images per frame
+            for index, tf_cam_image in enumerate(frame.images):
+                # 4x4 row major transform matrix that transforms
+                # 3d points from one frame to another.
+                SE3_flattened = np.array(tf_cam_image.pose.transform)
+                city_SE3_egovehicle = SE3_flattened.reshape(4, 4)
+                # in seconds
+                timestamp_s = tf_cam_image.pose_timestamp
+                timestamp_ns = round_to_micros(int(timestamp_s * 1e9))  # to nanoseconds
+                if args.save_poses:
+                    dump_pose(city_SE3_egovehicle, timestamp_ns, log_id, ARGO_WRITE_DIR)
+
+                if args.save_images:
+                    camera_name = CAMERA_NAMES[tf_cam_image.name]
+                    img = tf.image.decode_jpeg(tf_cam_image.image)
+                    new_img = undistort_image(
+                        np.asarray(img),
+                        frame.context.camera_calibrations,
+                        tf_cam_image.name,
+                    )
+                    img_save_fpath = f"{ARGO_WRITE_DIR}/{log_id}/{camera_name}/"
+                    img_save_fpath += f"{camera_name}_{timestamp_ns}.jpg"
+                    # assert not Path(img_save_fpath).exists()
+                    check_mkdir(str(Path(img_save_fpath).parent))
+                    imageio.imwrite(img_save_fpath, new_img)
+                    img_count += 1
+                    if img_count % 100 == 0:
+                        print(f"\tSaved {img_count}'th image for log = {log_id}")
+
+
+def undistort_image(img: np.ndarray, calib_data: Any, camera_name: int):
+    """Undistort the image from the Waymo dataset given camera calibration data"""
+    for camera_calib in calib_data:
+        if camera_calib.name == camera_name:
+            f_u, f_v, c_u, c_v, k1, k2, p1, p2, k3 = camera_calib.intrinsic
+            # k1, k2 and k3 are the tangential distortion coefficients
+            # p1, p2 are the radial distortion coefficients
+            camera_matrix = np.array([[f_u, 0, c_u], [0, f_v, c_v], [0, 0, 1]])
+            dist_coeffs = np.array([k1, k2, p1, p2, k3])
+            return cv2.undistort(img, camera_matrix, dist_coeffs)
+
+
+def form_calibration_json(calib_data):
```

Comment:
can we add the types for input and return arguments here?

Reply:
Addressed [here](https://github.com/johnwlambert/waymo_to_argoverse/pull/2/commits/4ce61dbfa76327be7e9627546897df849c7b8cff)